console/console-init/ui/src/modules/iot-device-detail/components/ConnectionGateway/ConnectionGateway.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { useParams } from "react-router";
+import {
+  Card,
+  Flex,
+  FlexItem,
+  CardBody,
+  Title,
+  Text,
+  Popover,
+  Button,
+  TextVariants,
+  CardTitle
+} from "@patternfly/react-core";
+import { OutlinedQuestionCircleIcon } from "@patternfly/react-icons";
+import { StyleSheet, css } from "aphrodite";
+export interface IConnectionGatewayProps {
+  deviceList?: any;
+  deviceGroup?: any;
+}
+const styles = StyleSheet.create({
+  card_body: {
+    marginBottom: 20
+  },
+  text_margin: {
+    marginLeft: 25
+  }
+});
+
+export const ConnectionGateway: React.FC<IConnectionGatewayProps> = ({
+  deviceList,
+  deviceGroup
+}) => {
+  const { projectname, namespace } = useParams();
+
+  return (
+    <Card className={css(styles.card_body)}>
+      <CardTitle>
+        <Title id="device-info-gateways-title" headingLevel="h1" size="2xl">
+          Connection Gateways&nbsp;&nbsp;
+          <Popover
+            enableFlip={false}
+            bodyContent={<div>Gateway Groups to which the device belongs</div>}
+            aria-label="Add gateway devices info popover"
+            closeBtnAriaLabel="Close Gateway Devices info popover"
+          >
+            <Button
+              variant="link"
+              icon={<OutlinedQuestionCircleIcon />}
+            ></Button>
+          </Popover>
+        </Title>
+      </CardTitle>
+      {deviceList.length > 0 && (
+        <text className={css(styles.text_margin)}>Gateway Devices</text>
+      )}
+      <CardBody>
+        <Flex>
+          {deviceList &&
+            deviceList.map((deviceId: string) => {
+              return (
+                <FlexItem span={2} key={deviceId}>
+                  <Link
+                    id="device-info-id-link"
+                    to={`/iot-projects/${namespace}/${projectname}/devices/${deviceId}/device-info`}
+                  >
+                    {deviceId}
+                  </Link>
+                </FlexItem>
+              );
+            })}
+        </Flex>
+        <br />
+        {deviceList.length > 0 && <text>Gateway Groups</text>}
+        <br />
+        <Flex>
+          {deviceGroup &&
+            deviceGroup.map((deviceId: string) => {
+              return (
+                <FlexItem span={2} key={deviceId}>
+                  <Link
+                    id="device-info-id-link"
+                    to={`/iot-projects/${namespace}/${projectname}/devices/${deviceId}/device-info`}
+                  >
+                    {deviceId}
+                  </Link>
+                </FlexItem>
+              );
+            })}
+        </Flex>
+        {deviceList?.length <= 0 && deviceGroup?.length <= 0 && (
```

Comment:
I think we don't need this logic. According to new wireframe if we don't have deviceList and deviceGroup card will not display. Please follow the wireframe.